src/runtime/native/StrPtr.cs

```diff
@@ -0,0 +1,73 @@
+#nullable enable
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Python.Runtime.Native
+{
+    [StructLayout(LayoutKind.Sequential)]
+    struct StrPtr : IDisposable
```

Comment:
Naming things :-D

Not sure about `NativeString`. It has connotations to native string type supported on the platform, but this is not it.

It might make sense to have a `StrPtr` type per encoding for clarity. Then signatures could enforce it. But maybe that would be overengineering.

Reply:
Do we use any encoding apart from UTF8? I thought all `FromString` functions nowadays used that.

Reply:
The main thing I find "problematic" is that this object is not a pure pointer, it has ownership over the buffer it points to.